### PR TITLE
[4006] Reactify "Reject" dialog

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/4006

### CODE REVIEW

File: ui/app/src/models/Item.ts [hunk 1]
```diff
@@ -27,6 +27,9 @@ interface BaseItem {
   disabled: boolean;
   localeCode: string;
   translationSourceId: string;
+  submitted: boolean;
+  scheduled: boolean;
+  deleted: boolean;
```
Comment (rart) [inline ui/app/src/models/Item.ts]: These props aren't in any of the [new items](https://app.swaggerhub.com/apis/craftercms/studio/3.1.7.1#/DetailedItem), are they? Remove unless they are.

File: ui/app/src/services/configuration.ts [hunk 2]
```diff
@@ -324,6 +324,12 @@ export function getHistory(site: string, path: string, environment: string, modu
   );
 }
 
+export function fetchCannedMessage(site: string, locale: string, type: string): Observable<string> {
+   return get(`/studio/api/1/services/api/1/site/get-canned-message.json?site=${site}&locale=${locale}&type=${type}`).pipe(
+     pluck('response')
+   )
```
Comment (rart) [inline ui/app/src/services/configuration.ts]: Semi missing. Do you have the reformat, organize and clean up code checks on?

Add Api1 error handler

File: ui/app/src/services/publishing.ts [hunk 3]
```diff
@@ -55,6 +55,21 @@ export function goLive(siteId: string, user: string, data): Observable<any> {
   );
 }
 
+export function reject(siteId: string, dependencies: string[], items: string[], reason: string, submissionComment: string): Observable<any> {
```
Comment (rart) [inline ui/app/src/services/publishing.ts]: Type missing

File: ui/app/src/utils/content.ts [hunk 4]
```diff
@@ -101,7 +101,10 @@ export function parseLegacyItemToSandBoxItem(item: LegacyItem | LegacyItem[]): S
     modifier: null,
     lastModifiedDate: null,
     commitId: null,
-    sizeInBytes: null
+    sizeInBytes: null,
+    submitted: item.isSubmitted,
+    scheduled: item.isScheduled,
+    deleted: item.isDeleted
```
Comment (rart) [inline ui/app/src/utils/content.ts]: Remove if not part of the new items

File: ui/app/src/components/Dialogs/RejectDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,420 @@
...
+    fetchDependencies(siteId, checkedItems).subscribe(
+      (response) => {
+        // api being used in legacy (/studio/api/1/services/api/1/dependency/get-dependencies.json)
+        // returns only softDependencies
+        const deps = response.softDependencies;
+
+        setApiState({ ...apiState, submitting: true });
+
+        reject(siteId, deps, checkedItems, rejectionReason, rejectionComment).subscribe(
+          () => {
```
Comment (rart) [inline ui/app/src/components/Dialogs/RejectDialog.tsx]: Compose the observables on the reject service. Do a single call here.

**Review by rart — CHANGES_REQUESTED** [review]
Please address comments.

**Review by rart — APPROVED** [review]

**Review by rart — CHANGES_REQUESTED** [review]
- Conflicts
- Is the dialog receiving Sandbox items from legacy? Who calls the reject function con common api. That's unlikely to be sandbox items already.

**Comment by jvega190** [comment]
@rart dialog was not receiving sandbox items from legacy, `parseLegacyItemToSandBoxItem` is now being used for the items to be sent properly (sandboxItems)

File: ui/app/src/components/ComponentMenu.tsx [hunk 6]
```diff
@@ -216,6 +223,15 @@ export default function ComponentMenu(props: ComponentMenuProps) {
             />
           </MenuItem>
         }
+        {
+          (item && (item.state === 5 || item.state === 6 || item.state === 2)) &&
```
Comment (rart) [inline ui/app/src/components/ComponentMenu.tsx]: I don't think the state will ever be a plain number. This is supposed to be a bit map.

File: ui/app/src/components/Dialogs/RejectDialog.tsx [hunk 7]
```diff
@@ -0,0 +1,407 @@
...
+  onDismiss?: StandardAction;
+  onRejectSuccess?: StandardAction;
+}
+
+// endregion
+
+const useStyles = makeStyles(() =>
+  createStyles({
+    itemsList: {
+      border: '1px solid #D8D8DC',
```
Comment (rart) [inline ui/app/src/components/Dialogs/RejectDialog.tsx]: Use color from palette

File: ui/app/src/components/Dialogs/RejectDialog.tsx [hunk 8]
```diff
@@ -0,0 +1,407 @@
...
+                  <MenuItem key={'Typos'} value={'Typos'}>
+                    <FormattedMessage id="rejectDialog.typos" defaultMessage="Typos" />
+                  </MenuItem>
+                  <MenuItem key={'BrokenLinks'} value={'BrokenLinks'}>
+                    <FormattedMessage id="rejectDialog.brokenLinks" defaultMessage="Broken Links" />
+                  </MenuItem>
+                  <MenuItem key={'NSOA'} value={'NSOA'}>
+                    <FormattedMessage
+                      id="rejectDialog.nsoa" defaultMessage="Needs Section Owner's Approval"
+                    />
```
Comment (rart) [inline ui/app/src/components/Dialogs/RejectDialog.tsx]: Don't think we need the keys here. 
Also, change the syntax so plain strings are double quoted and without the curly braces

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — APPROVED** [review]

**Pull request merged**